Libs/MRML/DisplayableManager/vtkMRMLSliceIntersectionDisplayableManager.cxx

```diff
@@ -0,0 +1,270 @@
+/*==============================================================================
+
+  Program: 3D Slicer
+
+  Portions (c) Ebatinca S.L., Las Palmas de Gran Canaria, Spain
+  All Rights Reserved.
+
+     This software is distributed WITHOUT ANY WARRANTY; without even
+     the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+     PURPOSE.  See the above copyright notice for more information.
+
+==============================================================================*/
+
+// MRMLDisplayableManager includes
+#include "vtkMRMLSliceIntersectionDisplayableManager.h"
+
+// MRML includes
+#include <vtkMRMLApplicationLogic.h>
+#include <vtkMRMLColorNode.h>
+#include <vtkMRMLCrosshairNode.h>
+#include <vtkMRMLDisplayNode.h>
+#include <vtkMRMLInteractionNode.h>
+#include <vtkMRMLLightBoxRendererManagerProxy.h>
+#include <vtkMRMLScene.h>
+#include <vtkMRMLSliceCompositeNode.h>
+#include <vtkMRMLSliceIntersectionInteractionWidget.h>
+#include <vtkMRMLSliceLogic.h>
+#include <vtkMRMLSliceNode.h>
+
+// VTK includes
+#include <vtkActor2D.h>
+#include <vtkCallbackCommand.h>
+#include <vtkCellArray.h>
+#include <vtkEventBroker.h>
+#include <vtkMatrix4x4.h>
+#include <vtkNew.h>
+#include <vtkObjectFactory.h>
+#include <vtkPickingManager.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataMapper2D.h>
+#include <vtkProp.h>
+#include <vtkProperty2D.h>
+#include <vtkRenderer.h>
+#include <vtkRenderWindow.h>
+#include <vtkRenderWindowInteractor.h>
+#include <vtkSmartPointer.h>
+#include <vtkWeakPointer.h>
+#include <vtkVersion.h>
+
+// STD includes
+#include <algorithm>
+#include <cassert>
+
+//---------------------------------------------------------------------------
+vtkStandardNewMacro(vtkMRMLSliceIntersectionDisplayableManager);
+
+//---------------------------------------------------------------------------
+class vtkMRMLSliceIntersectionDisplayableManager::vtkInternal
+{
+  public:
+    vtkInternal(vtkMRMLSliceIntersectionDisplayableManager* external);
+    ~vtkInternal();
+
+    vtkObserverManager* GetMRMLNodesObserverManager();
+    void Modified();
+
+    // Slice
+    vtkMRMLSliceNode* GetSliceNode();
+    void UpdateIntersectingSliceNodes();
+    // Crosshair
+    void SetCrosshairNode(vtkMRMLCrosshairNode* crosshairNode);
+
+    // Actors
+    void SetActor(vtkActor2D* prop) { Actor = prop; };
+
+    // Build the crosshair representation
+    void BuildCrosshair();
+
+    // Add a line to the crosshair in display coordinates (needs to be
+    // passed the points and cellArray to manipulate).
+    void AddCrosshairLine(vtkPoints* pts, vtkCellArray* cellArray,
+      int p1x, int p1y, int p2x, int p2y);
+
+    // Did crosshair position change?
+    bool DidCrosshairPositionChange();
```

Comment:
These functions (`DidCrosshairPositionChange` and `DidCrosshairPropertyChange`) already existed in the **vtkMRMLCrosshairDisplayableManager** class. I will remove them from the **vtkMRMLSliceIntersectionDisplayableManager** class since they're not currently used. 

I will update the name of those functions in the **vtkMRMLCrosshairDisplayableManager** based on your suggestions.

Reply:
Perfect!